### PR TITLE
Feat - WIP 247 - Proof of Anonymity

### PR DESCRIPTION
**Pull request by jl-ib**
### What does this PR do?

Add a dropdown menu to select between Anonymous or Username, if the Username is selected a pop-up is shown when submitting the question, warning the user that the username will be posted, and asking for a confirmation/cancel.

### Where should the reviewer start?

src/components/QuestionInput.js on the render function of the class, the Dropdown button was added and the SubmitWithUserModal modal was added.
src/Auth.js a function was added to get the profile's name, this function is used in QuestionInput.js.

### How should this be manually tested?
Once logged in, the user should be able to see a dropdown before the input field for the question, this dropdown has Anonymous as default and its value can be changed when clicked. If the username is selected and the user clicks the "Ask Question!" button, a pop-up should be shown, asking for a confirmation, if it is cancelled, the dropdown value and the question input are not reset and the question is not submitted, if the user submits the question, the values are reset to Anonymous, and the inputfield is reset to empty ' '. 

### What are the relevant tickets?
[WIP - 247 [FronEnd] Proof of Anonymity](https://wizeline.atlassian.net/browse/WIP-247)

### Screenshots
image
image

### CODE REVIEW

File: src/components/QuestionInput.js [hunk 1]
```diff
@@ -17,51 +21,83 @@ class QuestionInput extends Component {
...
     });
   }
 
+  handleChange = (event) => {
+    this.setState({ inputValue: event.target.value });
+  }
+
+  selectPostingAs = (value) => {
+    this.setState({ dropDownTitle: value });
+    this.state.isAnonymousChecked = (value === 'Anonymous');
```
Comment (alan-maldonado) [inline src/components/QuestionInput.js]: maybe it is better to rename this property to `isAnnonymous`

File: src/components/QuestionInput.js [hunk 2]
```diff
@@ -17,51 +21,83 @@ class QuestionInput extends Component {
...
       this.props.fireAlert('success', 'Success', 'Question successfully saved');
       this.props.fetchQuestions();
     });
   }
 
+  handleChange = (event) => {
+    this.setState({ inputValue: event.target.value });
+  }
+
+  selectPostingAs = (value) => {
```
Comment (alan-maldonado) [inline src/components/QuestionInput.js]: can we rename to `username` instead of value?

File: src/components/QuestionInput.js [hunk 3]
```diff
@@ -17,51 +21,83 @@ class QuestionInput extends Component {
...
+    this.setState({ dropDownTitle: value });
+    this.state.isAnonymousChecked = (value === 'Anonymous');
+  }
+
+
+  handleSubmit = (event) => {
+    event.preventDefault();
+
+    if (this.state.inputValue === '') {
+      this.props.fireAlert('warning', 'Warning', 'Question Needed to Post');
```
Comment (alan-maldonado) [inline src/components/QuestionInput.js]: you can change this condition to if (!this.state.inputValue) { ... } to validate null or undefined also.

File: src/components/QuestionInput.js [hunk 4]
```diff
@@ -17,51 +21,83 @@ class QuestionInput extends Component {
...
+
+    if (!this.state.isAnonymousChecked) {
+      this.openSubmitWithUserModal();
+      return;
+    }
+    this.handleSubmitWithUserModalSuccess();
+  }
+
   render() {
     return (
```
Comment (alan-maldonado) [inline src/components/QuestionInput.js]: I think you could get the username here.
`const username = getProfile().name`

File: src/auth.js [hunk 5]
```diff
@@ -131,6 +131,10 @@ export function checkForAdminStatus() {
   return false;
 }
 
+export function getProfileName() {
+  return getProfile().name;
```
Comment (alan-maldonado) [inline src/auth.js]: Is this function worth it to be created?
It is just getting a property of another function. I think you could use destructing for example in other sections of the code.

const { name } = getProfile()
or if you want an alias
const { name: username } = getProfile()

Comment (imarban) [inline src/auth.js]: 💯

Comment (jl-ib) [inline src/auth.js]: Ok 😄 I deleted the function and made getProfile() exportable. I already pushed the rest of the changes as well.

File: src/components/QuestionInput.js [hunk 6]
```diff
@@ -16,52 +20,85 @@ class QuestionInput extends Component {
...
+      dropDownTitle: 'Anonymous',
+      showSubmitWithUserModal: false,
+      closed: false,
     };
   }
 
-  handleChange = (event) => {
-    this.setState({ inputValue: event.target.value });
-  }
+  resetForm = {
```
Comment (rominavarela) [inline src/components/QuestionInput.js]: `reset` implies an action. I suggest to convert this into a function the same way `openSubmitWithUserModal` is

**Pull request merged**